File: src/main/java/edu/cornell/cs/apl/viaduct/imp/ast/values/IntegerValue.java
package edu.cornell.cs.apl.viaduct.imp.ast.values;

import com.google.auto.value.AutoValue;
import edu.cornell.cs.apl.viaduct.imp.ast.types.ImpBaseType;
import edu.cornell.cs.apl.viaduct.imp.ast.types.IntegerType;

/** Integer literal. */
@AutoValue
public abstract class IntegerValue implements ImpValue {
  public static IntegerValue create(int value) {
    return new AutoValue_IntegerValue(value);
  }

  public abstract int getValue();

  @Override
  public final ImpBaseType getType() {
    return IntegerType.create();
  }

  @Override
  public final String toString() {
    return Integer.toString(this.getValue());
  }
}

File: src/main/kotlin/edu/cornell/cs/apl/viaduct/syntax/values/IntegerValue.kt
package edu.cornell.cs.apl.viaduct.syntax.values

import edu.cornell.cs.apl.viaduct.syntax.types.IntegerType
import edu.cornell.cs.apl.viaduct.syntax.types.ValueType

/** An integer. */
data class IntegerValue(val value: Int) : Value() {
    override val type: ValueType
        get() = IntegerType

    override fun toString(): String {
        return value.toString()
    }
}
